web/access_control.go

```diff
@@ -78,6 +78,54 @@ func (s *AutograderService) isEnrolled(userID, courseID uint64) bool {
 	})
 }
 
+// isValidSubmission returns true if submitter has active course enrollment or
+// if submitting group belongs to the given course
+func (s *AutograderService) isValidSubmissionRequest(submission *pb.SubmissionRequest) bool {
+	if !submission.IsValid() {
```

Comment:
isn’t the IsValid() method called automatically by the interceptor

Reply:
Yes, when the struct comes from the client, it absolutely does. But the method should work for locally created `SubmissionRequest` structs as well.